examples/dnn_semantic_segmentation_train_ex.cpp

```diff
@@ -0,0 +1,317 @@
+{
+    return find_voc2012_class(
+        [&rgb_label](const Voc2012class& voc2012class) {
+            return rgb_label == voc2012class.rgb_label;
+        }
+    );
+}
+
+inline uint16_t rgb_label_to_index_label(const dlib::rgb_pixel& rgb_label)
+{
+    return find_voc2012_class(rgb_label).index;
+}
+
+void rgb_label_image_to_index_label_image(const dlib::matrix<dlib::rgb_pixel>& rgb_label_image, dlib::matrix<uint16_t>& index_label_image)
+{
+    const long nr = rgb_label_image.nr();
+    const long nc = rgb_label_image.nc();
+
+    index_label_image.set_size(nr, nc);
+
+    for (long r = 0; r < nr; ++r) {
+        for (long c = 0; c < nc; ++c) {
+            index_label_image(r, c) = rgb_label_to_index_label(rgb_label_image(r, c));
+        }
+    }
+}
+
+// ----------------------------------------------------------------------------------------
+
+double calculate_accuracy(anet_type& anet, const std::vector<image_info>& dataset)
+{
+    int num_right = 0;
+    int num_wrong = 0;
+
+    matrix<rgb_pixel> input_image;
+    matrix<rgb_pixel> rgb_label_image;
+    matrix<uint16_t> index_label_image;
+
+    for (const auto& image_info : dataset) {
+        load_image(input_image, image_info.image_filename);
+        load_image(rgb_label_image, image_info.label_filename);
+
+        matrix<uint16_t> net_output = anet(input_image);
+
+        rgb_label_image_to_index_label_image(rgb_label_image, index_label_image);
+
+        const long nr = index_label_image.nr();
+        const long nc = index_label_image.nc();
+
+        for (long r = 0; r < nr; ++r) {
+            for (long c = 0; c < nc; ++c) {
+                const uint16_t truth = index_label_image(r, c);
+                if (truth != dlib::loss_multiclass_log_per_pixel_::label_to_ignore) {
+                    const uint16_t prediction = net_output(r, c);
+                    if (prediction == truth) {
+                        ++num_right;
+                    }
+                    else {
+                        ++num_wrong;
+                    }
+                }
+            }
+        }
+    }
+
+    return num_right / static_cast<double>(num_right + num_wrong);
+}
+
+// ----------------------------------------------------------------------------------------
+
+int main(int argc, char** argv) try
+{
+    if (argc != 2)
+    {
+        cout << "To run this program you need a copy of the PASCAL VOC2012 dataset." << endl;
+        cout << endl;
+        cout << "You call this program like this: " << endl;
+        cout << "./dnn_semantic_segmentation_train_ex /path/to/VOC2012" << endl;
+        return 1;
+    }
+
+    cout << "\nSCANNING PASCAL VOC2012 DATASET\n" << endl;
+
+    const auto listing = get_pascal_voc2012_train_listing(argv[1]);
+    cout << "images in dataset: " << listing.size() << endl;
+    if (listing.size() == 0)
+    {
+        cout << "Didn't find the VOC2012 dataset. " << endl;
+        return 1;
+    }
+        
+    set_dnn_prefer_smallest_algorithms();
```

Comment:
Do you still need this?  I think the current version of dlib should in general not require this anymore.  Like it shouldn't ever matter much but it's slower.

Reply:
Probably not – removed.